decoder/LAVAudio/LAVAudio.cpp

```diff
@@ -1742,9 +1750,13 @@ static int check_mpegaudio_header(uint8_t *buf, uint32_t *retheader)
 HRESULT CLAVAudio::ResyncMPEGAudio()
 {
   uint8_t *buf = m_buff.Ptr();
-  int size = m_buff.GetCount();
+  DWORD size = m_buff.GetCount();
+  if (size < 3) {
+    DbgLog((LOG_TRACE, 10, L"CLAVAudio::ResyncMPEGAudio(): Buffer too short to perform resync"));
+    return S_FALSE;
+  }
 
-  for (int i = 0; i < (size - 3); i++)
+  for (unsigned int i = 0; i < (size - 3); ++i)
```

Comment:
True, but I wasn't sure if changing the datatype of `i` beyond just its signedness could introduce any unexpected behavior given the code in the loop, so I settled on just fixing the signedness. However, if you think there's no impact, then yes, `DWORD` is the way to go!